design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -0,0 +1,211 @@
+
+The proposed solution is deprecation of `customerOrders` query and `orders` field to the `customer` query:
+
+```graphql
+# query to return a list of all customer orders
+type Customer {
+    orders (
+        filter: CustomerOrdersFilterInput
+        currentPage: Int = 1 # current page of the customer order list. default is 1.
+        pageSize: Int = 20 # page size for the customer orders list. default is 20.
+    ): CustomerOrders
+}
+
+input CustomerOrdersFilterInput {
+    order_number: String!
+}
+
+# collection of customer orders.
+type CustomerOrders {
+    items: [CustomerOrder]! # collection of customer orders that contains individual order details.
+    page_info: SearchResultPageInfo
+    total_count: Int
+}
+```
+
+> Right now, we don't introduce the filter input for such entities like invoice, credit memo, shipment as all these entities are related to the same order and GraphQL resolvers anyway at first resolve the order and after that other child entities. But, in future, such filter input can be introduced as optional argument.
+
+## Order Type Schema
+
+The proposed type for the customer order might look like:
+
+```graphql
+# order details
+type CustomerOrder {
+    order_date: String! # date when the order was placed.
+    status: String! # current status of the order.
+    order_number: String! # order number.
+    items: [OrderItem] # collection of all the items purchased
+    prices: SalesPricesInterface! # prices details for the order.
+    invoices: [Invoice]! # invoice list for the order.
+    credit_memos: [CreditMemo]! # credit memo list for the order.
+    shipments: [OrderShipment]! # shipping list for the order.
+    payment_methods: [PaymentMethod]! # payment details for the order.
+}
+```
+
+> The order `status` should be filtered in the same way as for Luma via `Order Status` and `Visible On Storefront` configuration 
+
+### Order Item
+
+The order items will be presented as separate interface which will have multiple implementations for invoice, shipment and credit memo types.
+
+```graphql
+interface SalesItemInterface {
+    name: String # name of the base product.
+    sku: String! # sku of the base product.
+    url: String # url of the base product.
+    sale_price: Float! # sale price for the base product including all the child products and selected options.
+    discounts: [Discount] # final discount information for the base product including discounts on options and child products.
+    selected_options: [String!] # selected options for the base product. for e.g color, size etc.
+    entered_options: [SalesItemOption] # entered option for the base product. for e.g logo image etc.
+}
+
+type SalesItemOption {
+    id: String!
+    value: String!
+}
+```
+
+The `SalesItemInterface` will be implemented by the following types:
+
+```graphql
+# Order Product implementation of OrderProductInterface
+type OrderItem implements SalesItemInterface {
+    qty_ordered: Float! # number of items items
```

Comment:
Please rename all `qty` to quantity across the PR. We have a task in backlog to do this for all existing fields too.